### DISCUSSION TITLE
HyperGraphQL service with nautilus/ gateway

### DISCUSSION THREAD

**Discussion by ABIDRISA** [nautilus/gateway#160] [Q&A] [answered]
Hello, 
i am searching for a way to integrate a triplestore RDF with a MongoDB database, i found HyperGraphQL interesting as a connector to the triplestore RDF, could the nautilus gateway accept a service of HyperGraphQL as one of his services ! 
thank Youu.

**Comment by JohnStarich** [comment, marked as answer]
Hi @ABIDRISA, I think the answer for "Can I use X GraphQL service in nautilus/gateway?" is generally yes. One limitation might be it's HTTP-only today, but that's a non-issue 99% of the time.

Is there a specific feature or issue you're interested in?

**Reply by JohnStarich** [reply]
> Hello,
> My fist issue is that the HyperGraphQL service is started in another project and i wont to access to it from the nautilus gateway with its URL (ex : http://localhost:8000/graphql), but here i see that the services are in the same project, could the nautilus gateway extract the schéma alone from the URL of the service GraphQL?. the second is that the HyperGraphQL service uses JSON-LD context so the response to queries is accomplishes by (@ context) at the end, how did this could be traited in nautilus gateway, and what changer do i have to do in this case?
> thank you

(Copying here into the same thread)

Just from reading the docs, it looks like it’s a bog-standard GraphQL query request over HTTP. As long as that endpoint also supports introspection queries (pretty common) then the gateway should be able to pick up the schema okay.

The JSON-LD response format is unusual. I don’t think the gateway will recognize and return those extra `@context` fields, but the normal fields might work.

It seems like you’ll need to try it out to know for sure.

**Comment by ABIDRISA** [comment]
Hello, 
My fist issue is that the HyperGraphQL service is started in another project and i wont to access to it from the nautilus gateway with its URL (ex : http://localhost:8000/graphql), but here i see that the services are in the same project, could the nautilus gateway extract the schéma alone from the URL of the service GraphQL?. the second is that the HyperGraphQL service uses JSON-LD context so the response to queries is accomplishes by (@ context) at the end, how did this could be traited in nautilus gateway, and what changer do i have to do in this case? 
thank you
